DAQ/src/StrawRecoFromArtdaqFragments_module.cc

```diff
@@ -0,0 +1,297 @@
+        //   break;
+        // }
+
+        for (size_t ii = 0; ii < contf.block_count(); ++ii) {
+          containerFragments.push_back(contf[ii]);
+          fragments.push_back(*containerFragments.back());
+        }
+      }
+    }
+    else {
+//-----------------------------------------------------------------------------
+// the 'handle' handles a list of artdaq fragments
+// each artdaq fragment corresponds to a single DTC, or a plane
+// loop over them
+//-----------------------------------------------------------------------------
+      int nfrag = handle->size();
+      for (int ifrag=0; ifrag<nfrag; ifrag++) {
+        const auto& frag = handle->at(ifrag);
+        uint8_t* fdata  = (uint8_t*) (frag.dataBegin());
+        TLOG(TLVL_DEBUG) << "-- fragment number:" << ifrag;
+//-----------------------------------------------------------------------------
+// skip non-tracker fragments
+//-----------------------------------------------------------------------------
+        DTCLib::DTC_SubEventHeader* seh = (DTCLib::DTC_SubEventHeader*) fdata;
+        if (seh->source_subsystem != DTCLib::DTC_Subsystem::DTC_Subsystem_Tracker) continue;
+        int dtc_id  = seh->source_dtc_id;
+//-----------------------------------------------------------------------------
+// this is a tracker DTC fragment, loop over the ROCs
+//-----------------------------------------------------------------------------
+        ushort*  buf          = (ushort*) fdata;
+        int      nbytes       = buf[0]; // frag.dataSizeBytes() includes extra 0x20
+        uint8_t* roc_data     = fdata+sizeof(*seh);
+        uint8_t* last_address = fdata+nbytes;     //
+
+        while (roc_data < last_address) {
+          RocDataHeaderPacket_t* rdh = (RocDataHeaderPacket_t*) roc_data;
+
+          int nhits = rdh->packetCount/(nADCPackets_+1); // nADCPackets_ defaults to 1
+
+          int link_id = rdh->linkID;
+          TLOG(TLVL_DEBUG) << "DTC:" << dtc_id << " ROC:" << link_id
+                           << " offset:" << std::hex
+                           << " nhits:" << std::dec << nhits;
+
+          for (int ihit=0; ihit<nhits; ihit++) {
+//-----------------------------------------------------------------------------
+// first packet, 16 bytes, or 8 ushort's is the data header packet
+//-----------------------------------------------------------------------------
+            mu2e::TrackerDataDecoder::TrackerDataPacket* hit ;
+            int offset          = (ihit*np_per_hit+1)*packet_size;   // in bytes
+            hit = (mu2e::TrackerDataDecoder::TrackerDataPacket*) (roc_data+offset);
+//-----------------------------------------------------------------------------
+// unpack hit and fill the StrawDigiCollection
+// HV side has an extra bit in teh straw ID
+//-----------------------------------------------------------------------------
+            uint16_t straw_id = hit->StrawIndex;
+            if (straw_id >= 0x80) straw_id = straw_id - 0x80;
```

Comment:
This comes from just how the panel id registers are set on powerup right now - if those are written to correctly by the DAQ before taking data this fix is not needed

Reply:
fixed